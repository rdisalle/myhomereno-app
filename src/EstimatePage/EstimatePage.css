.EstimatePage_list {
    list-style-type: none;
    padding-left: 0px;
    padding-right: 0px;
}