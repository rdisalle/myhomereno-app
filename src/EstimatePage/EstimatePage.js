import React, { Component } from 'react';
import './EstimatePage.css';
import { Link } from 'react-router-dom';
import MyHomeRenoContext from '../MyHomeRenoContext';
import config from '../config';
import PropTypes from 'prop-types';
import format from 'date-fns/format';

class EstimatePage extends Component {
    static defaultProps = {
        name: '',
        contractor_name: '',
        price: '',
        details: '',
        total_time: '',
      };

    static contextType = MyHomeRenoContext;

    handleClickDelete = e => {
        e.preventDefault()
        const estimateId = this.props.id;
    
        fetch(`${config.API_ENDPOINT}/api/estimates/${estimateId}`, {
          method: 'DELETE',
          headers: {
            'content-type': 'application/json'
          },
        })
          .then(res => {
            if (!res.ok)
              return res.json().then(e => Promise.reject(e));
            return;
          })
          .then(() => {
            this.context.deleteEstimate(estimateId);
          })
          .catch(error => {
            console.error({ error });
          });
      };

  render() {
    const { id, name, contractor_name, price, details, total_time, date_created} = this.props;
    return (
        <div className="EstimatePage">
            <div className="EstimateInfo">
            <ul className="EstimatePage_list">
                <li className="EstimatePage_name">
                {name}
                </li>
                <li className="EstimatePage_info">
                Contractor: {' '} {contractor_name}
                </li>
                <li className="EstimatePage_info">
                Price: {' '} {price}
                </li>
                <li className="EstimatePage_info">
                {details} 
                </li>
                <li className="EstimatePage_info">
                Time Frame: {' '} {total_time}
                </li>
                <li>
                <div className="Estimate_dates">
                <div className="Estimate_dates-created">
                    Created:
                    {' '}
                    <span className="Date">
                    {format(new Date(date_created), "LLLL do yyyy")}
                    </span>
                </div>
                </div>
                </li>
            </ul>
            </div>
            <div className="EstimatePage_command">
             <Link className="EstimatePage_link" to={`/edit-estimate/${id}`}>
                    Edit Estimate
            </Link>
            </div>
            <button
                className="Estimate_Delete"
                type='button'
                onClick={this.handleClickDelete}
            > 
            Delete
            </button>
      </div>
    );   
  };
};

EstimatePage.propTypes = {
    id: PropTypes.number,
    onDeleteNote: PropTypes.func,
    name: PropTypes.string.isRequired,
    contractor_name: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    details: PropTypes.string.isRequired,
    total_time: PropTypes.string.isRequired,
  };

export default EstimatePage;

