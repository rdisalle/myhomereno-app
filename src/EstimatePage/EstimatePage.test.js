import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EstimatePage from './EstimatePage'

describe(`EstimatePage component`, () => {
  const props = {
    id: 1,
    name: 'test-class-name',
    date_created: new Date('2020-01-22T16:28:32.615Z'),
  }

  it('renders the EstimatePage given props', () => {
    const wrapper = shallow(<EstimatePage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
