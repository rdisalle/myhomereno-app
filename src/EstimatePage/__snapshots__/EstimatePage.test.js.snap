// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstimatePage component renders the EstimatePage given props 1`] = `
<div
  className="EstimatePage"
>
  <div
    className="EstimateInfo"
  >
    <ul
      className="EstimatePage_list"
    >
      <li
        className="EstimatePage_name"
      >
        test-class-name
      </li>
      <li
        className="EstimatePage_info"
      >
        Contractor: 
         
         
      </li>
      <li
        className="EstimatePage_info"
      >
        Price: 
         
         
      </li>
      <li
        className="EstimatePage_info"
      />
      <li
        className="EstimatePage_info"
      >
        Time Frame: 
         
         
      </li>
      <li>
        <div
          className="Estimate_dates"
        >
          <div
            className="Estimate_dates-created"
          >
            Created:
             
            <span
              className="Date"
            >
              January 22nd 2020
            </span>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="EstimatePage_command"
  >
    <Link
      className="EstimatePage_link"
      to="/edit-estimate/1"
    >
      Edit Estimate
    </Link>
  </div>
  <button
    className="Estimate_Delete"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</div>
`;
